palantirnet/solr-faceted-search-react
# UMUS-28 search snippet highlighting + UMUS-102 no wildcard queries

## Tickets
* [UMUS-28](https://palantir.atlassian.net/browse/UMUS-28)
* [UMUS-102](https://palantir.atlassian.net/browse/UMUS-102)

## Description 
This PR updates our fork of the solr-faceted-search-react codebase to:
1. Enable a `mainQueryField` flag to be set, allowing an existing search field to be used to populate the main query param ("q") instead of being added as a filter list field ("fl")
1. Enable empty search results (vs always sending wildcard queries) by adding a `buildMainQuery` method which populates the "q" param based on the presence and value of the `mainQueryField` flag
1. Enable search result snippet highlighting by adding a `buildHighlight` method which populates enables solr highlighting and, when passed, adds subsequent `hl` params
1. Add and update any necessary unit tests

## To test
* See functional test script in https://github.com/palantirnet/umus/compare/umus-28-result-snippet-highlight to verify this work
* Optional: you can clone this repo, check out this branch, and run `npm run test` to verify passing unit tests
